modules/configloader/src/BoardConfSchema.cxx

```diff
@@ -0,0 +1,55 @@
+/*
+ *  BoardConfSchema.cxx
+ *  Copyright 2020 ItJustWorksTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+namespace smce {
+
+const char* board_conf_schema = R"(
```

Comment:
I noticed that you sometimes use CamelCase, and sometimes leading underscores is there any reason behind that?

Reply:
Casing is simple:
Macros: SHOUTING_SNAKE_CASE
Types: PascalCase
namspaces, functions, variables: snake_case
enumerators can either be like macros or like namespaces

If there are any leading underscores, it's definitely a mistake, unless it's under `include/stdpolyfills` since these extend the [compiler implementation](https://eel.is/c++draft/lex.name#3)

Reply:
You'll pretty much never, ever see me declare any identifiers in pascalCase, unless I have to implement an existing interface which follows that convention, like the Arduino API, or need to override virtual functions in a library which also follows that convention, such as Urho3D.